src/psyclone/psyir/symbols/symbol.py

```diff
@@ -89,20 +212,35 @@ class Visibility(Enum):
     # given an explicit visibility.
     DEFAULT_VISIBILITY = Visibility.PUBLIC
 
-    def __init__(self, name, visibility=DEFAULT_VISIBILITY):
+    def __init__(self, name, visibility=DEFAULT_VISIBILITY, interface=None):
```

Comment:
You also need to add the interface parameter to all sub-classes inits e.g. ContainerSymbol even if some combinations don't make sense for Fortran, this may need some extra checks in the backend gen_declarations.

Reply:
All symbols except ContainerSymbol already do this. For ContainerSymbol the class only allows one type of interface. I've therefore omitted to add interface as an option, but instead pass the fixed value to the Symbol base class. This seems sensible to me seeing as the class explicitly forbids other options at the moment. I suggest we allow other options when more than one is valid. However, I will change it if you insist :-)